web_generator/lib/src/interop_gen/transform/transformer.dart

```diff
@@ -72,6 +79,537 @@ class Transformer {
     nodes.add(node);
   }
 
+  InterfaceDeclaration _transformInterface(TSInterfaceDeclaration interface) {
+    // get name
+    final name = interface.name.text;
+
+    final modifiers = interface.modifiers?.toDart;
+    final isExported = modifiers?.any((m) {
+          return m.kind == TSSyntaxKind.ExportKeyword;
+        }) ??
+        false;
+
+    // get heritage clauses
+    // get heritage clauses
+    final heritageClauses = interface.heritageClauses?.toDart ?? [];
+
+    final extendees = <Type>[];
+    final implementees = <Type>[];
+
+    for (final clause in heritageClauses) {
+      if (clause.token == TSSyntaxKind.ExtendsKeyword) {
+        // extends
+        extendees.addAll(
+            clause.types.toDart.map(_transformTypeExpressionWithTypeArguments));
+      } else {
+        implementees.addAll(
+            clause.types.toDart.map(_transformTypeExpressionWithTypeArguments));
+      }
+    }
+
+    final (:id, name: dartName) = namer.makeUnique(name, 'interface');
+    final typeParams = interface.typeParameters?.toDart;
+
+    final outputInterface = InterfaceDeclaration(
+        name: name,
+        dartName: dartName,
+        id: id,
+        exported: isExported,
+        typeParameters:
+            typeParams?.map(_transformTypeParamDeclaration).toList() ?? [],
+        extendedTypes: extendees,
+        methods: List.empty(growable: true),
```

Comment:
Why not just `[]` here and elsewhere? Also, we can exclude these parameters from the constructor (and just have field initializers) if they're always going to be initialized to empty lists.

https://api.dart.dev/dart-core/List/List.empty.html

> If growable is true, the list is growable and equivalent to <E>[].